#### *ISEL - DEI - LEIM*
## Aprendizagem Automática [T52D]
### **Trabalho Laboratorial 1:** *Pima Indians Diabetes Dataset*

- João Madeira ($48630$),
- Renata Góis ($51038$),
- Bruno Pereira ($51811$)

**Docente responsável:** Prof. Gonçalo Xufre Silva

In [1]:
import pickle
import numpy as np
import matplotlib.pyplot as plt

from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC

#### **1.** Análise da base de dados
---

In [2]:
with open("pimaDiabetes.p", "rb") as f:
    dataset = pickle.load(f)
print("Tipo:", type(dataset))

X = dataset.data   # features
y = dataset.target # labels
print("\nDimensão de X:", X.shape)
print("Dimensão de y:", y.shape)

Tipo: <class 'sklearn.utils._bunch.Bunch'>

Dimensão de X: (768, 8)
Dimensão de y: (768,)


O *Pima Indians Diabetes Dataset*, originalmente desenvolvido pelo Instituto Nacional de Diabetes e Doenças
Digestivas e Renais dos Estados Unidos da America, contém informações sobre $768$ mulheres de uma das tribos indígenas desse país. Devido a uma alta taxa de incidencia de diabetes, a população dos índios Pima tem estado sobre contínua vigilância pelo Instituto desde 1965, cada residente tendo sido submetido a um exame padronizado cada dois anos. 

Nos dados disponibilizados, $500$ mulheres não têm diabetes e $268$ sim. \
A informação recolhida consiste nas seguintes $8$ características:

In [3]:
features = getattr(dataset, "feature_names", None)

means = np.mean(X, axis=0);stds = np.std(X, axis=0); mins = np.min(X, axis=0); maxs = np.max(X, axis=0)
for i, name in enumerate(features):
    print(f"{i+1}. {name:80s}  min={mins[i]:0.3f}  max={maxs[i]:0.2f}  média={means[i]:0.2f}  var={stds[i]:0.2f}")

1. Number of times pregnant                                                          min=0.000  max=17.00  média=3.85  var=3.37
2. Plasma glucose concentration at 2 Hours in an oral glucose tolerance test (GTIT)  min=0.000  max=199.00  média=120.89  var=31.95
3. Diastolic Blood Pressure (mm Hg)                                                  min=0.000  max=122.00  média=69.11  var=19.34
4. Triceps skin fold thickness (mm)                                                  min=0.000  max=99.00  média=20.54  var=15.94
5. 2-Hour Serum insulin (µh/ml)                                                      min=0.000  max=846.00  média=79.80  var=115.17
6. Body mass index [weight in kg/(Height in m)]                                      min=0.000  max=67.10  média=31.99  var=7.88
7. Diabetes pedigree function                                                        min=0.078  max=2.42  média=0.47  var=0.33
8. Age (years)                                                                       min=21